### PR TITLE
feat(telemetry)_: add metrics for message reliability

### PR DESCRIPTION
**Pull request by adklempner**
Adds metrics for missed messages, delivery confirmation, and peer count by shard/origin.

Tracks additional events in telemetry:
- when a message is successfully delivered
- when a check against a store node detects missed messages (with a separate count for if it was a message relevant to the user, as determined if the filter matches)
- when a dial to a peer fails, what kind of failure was encountered

Important changes:
- Depends on https://github.com/status-im/telemetry/pull/69

Dogfooding PR: https://github.com/status-im/status-desktop/pull/16540

### CODE REVIEW

File: telemetry/client.go [hunk 1]
```diff
@@ -18,26 +18,31 @@ import (
...
+	ErrorSendingEnvelopeMetric     TelemetryType = "ErrorSendingEnvelope"
+	PeerCountMetric                TelemetryType = "PeerCount"
+	PeerConnFailuresMetric         TelemetryType = "PeerConnFailure"
+	MessageCheckSuccessMetric      TelemetryType = "MessageCheckSuccess"
+	MessageCheckFailureMetric      TelemetryType = "MessageCheckFailure"
+	PeerCountByShardMetric         TelemetryType = "PeerCountByShard"
+	PeerCountByOriginMetric        TelemetryType = "PeerCountByOrigin"
+	DialFailureMetric              TelemetryType = "DialFailure"
+	MissedMessageMetric            TelemetryType = "MissedMessages"
+	MissedRelevantMessageMetric    TelemetryType = "MissedRelevantMessages"
```
Comment (qfrank) [inline telemetry/client.go]: It would be great to add some comments(like how it works) on these MissedXXX types

Comment (adklempner) [reply telemetry/client.go]: Added comments for each type indicating what is being tracked

**Review by qfrank — APPROVED** [review]
LGTM

File: telemetry/client.go [hunk 2]
```diff
@@ -467,6 +549,46 @@ func (c *Client) ProcessPeerCountByOrigin(peerCountByOrigin PeerCountByOrigin) *
 	return &jsonRawMessage
 }
 
+func (c *Client) ProcessDialFailure(dialFailure DialFailure) *json.RawMessage {
+	postBody := c.commonPostBody()
+	postBody["errorType"] = dialFailure.ErrorType
+	postBody["errorMsg"] = dialFailure.ErrorMsg
+	postBody["protocols"] = dialFailure.Protocols
+	body, _ := json.Marshal(postBody)
```
Comment (igor-sirotin) [inline telemetry/client.go]: Why ignore the error? 🤔

File: wakuv2/common/helpers.go [hunk 3]
```diff
@@ -110,3 +114,124 @@ func ValidateDataIntegrity(k []byte, expectedSize int) bool {
...
+	dialAttempts := strings.Split(strings.TrimSpace(match[1]), "\n")
+
+	// Regular expression to extract multiaddr and error message
+	reAttempt := regexp.MustCompile(`\[(.*?)\]\s*(.*)`)
+
+	var dialErrors []DialError
+	for _, attempt := range dialAttempts {
+		attempt = strings.TrimSpace(strings.Trim(attempt, "* "))
+		matches := reAttempt.FindStringSubmatch(attempt)
+		if len(matches) == 3 {
```
Comment (igor-sirotin) [inline wakuv2/common/helpers.go]:
```suggestion
		if len(matches) != 3 {
		    continue
		}
		// ...
```

File: wakuv2/common/helpers.go [hunk 4]
```diff
@@ -110,3 +114,124 @@ func ValidateDataIntegrity(k []byte, expectedSize int) bool {
...
+		"Relay No Reservation",
+		"Security Negotiation Failed",
+		"Concurrent Dial Succeeded",
+		"Concurrent Dial Failed",
+		"Connections Per IP Limit Exceeded",
+		"Stream Reset",
+		"Relay Resource Limit Exceeded",
+		"Error Opening Hop Stream to Relay",
+		"Dial Backoff",
+	}[det]
```
Comment (igor-sirotin) [inline wakuv2/common/helpers.go:199]: Though I like the hack, I think `map[string]string` is a more reliable solution in terms of potential changes. And the performance should remain the same.

**Review by igor-sirotin — APPROVED** [review]

**Pull request merged**